muttlib.c

```diff
@@ -995,8 +995,9 @@ void mutt_expando_format(char *buf, size_t buflen, size_t col, int cols, const c
         /* eat the format string */
         cp = prefix;
         count = 0;
-        while ((count < sizeof(prefix)) && (isdigit((unsigned char) *src) || (*src == '.') ||
-                                            (*src == '-') || (*src == '=')))
+        while ((count < (sizeof(prefix) - 1)) && (*src != '\0') &&
+               (isdigit((unsigned char) *src) || (*src == '.') ||
+                (*src == '-') || (*src == '=')))
```

Comment:
I would avoid any extra checks and try to simplify the logic as much as possible.

```c
        while ((count < (sizeof(prefix) - 1)) && (isdigit(*src) || strchr(".-=", *src)))
```

or even

```c
        while ((count < (sizeof(prefix) - 1)) && strchr("0123456789.-=", *src))
```

Reply:
Nice!  Done.